### DISCUSSION TITLE
ListingFile upload validation valid name error

### DISCUSSION THREAD

**Discussion by erikpolder** [etsy/open-api#1129] [Q&A] [answered]
I have a strange error. I can upload images to a draft listing so my application works as expected. Any ideas how to solve this validation issue?

Whenever I try to add a ListingFile I get the following response
`Received HTTP status code [400] with error "A valid name must be provided with a new file.".`

My payload looks like this:
```
[
  "file" => "filename.txt"
  "name" => "filename.txt"
  "rank" => 1
]
```
(obviously correct path to the file, similar to image upload)

[The documentation](https://developers.etsy.com/documentation/reference/#operation/uploadListingFile)

**Comment by turboturtleD** [comment]
Did you try it without using the .txt extension for the name parameter?

**Reply by erikpolder** [reply]
Hi! Thanks for your efforts in helping me.

Yes. It seems not to matter what I provide in the name field =(

```
[
  "file" => "filename.txt"
  "name" => "This is a test"
  "rank" => 1
]
```
Received HTTP status code [400] with error "A valid name must be provided with a new file."._

**Reply by turboturtleD** [reply]
Did you try it without spaces?

**Comment by erikpolder** [comment, marked as answer]
Resolution:
I was using rhysnhall/etsy-php-sdk which does not seem to work even when tweaking the source. I switched to regular Guzzle client and got my file to Etsy.

Thank you @turboturtleD  for spending time and trying to help.
